File: Frontend/src/app/viewModels/requests/ChangePasswordVM.ts
import { toJSONString } from "../interfaces/toJSONString";

/**
 * View Model class for Change Password request
 */
export class ChangePasswordVM implements toJSONString{

    /**
     * Construct's an instance of {@link ChangePasswordVM}
     * 
     * @param password new password
     */
    constructor(
        public password: string
    ) { }

    /**
     * Converts the class into a string representation of a json object 
     * 
     * @returns string representation of the object
     */
    public toJSONString(): string {
        return JSON.stringify({"password":this.password})
    }
}

File: lei_lds2122_grupo2-main/Backend/Data/ViewModels/Request/ChangePasswordVM.cs
namespace Backend.Data.ViewModels
{
    /// <summary>
    /// View Model class for Change Password reqeust
    /// </summary>
    public class ChangePasswordVM
    {
        public string Password { get; set; }

        /// <summary>
        /// Class' constructor method
        /// </summary>
        /// <param name="password">The new password</param>
        public ChangePasswordVM(String password){
            this.Password = password;
        }
    }
}